Customer.java

```diff
@@ -3,59 +3,73 @@
+
+    private int age;
+    private String name;
+    private boolean auth = false;  //authentication indicator. Whether customer is authenticated or not. All methods must check for authentication.
     public String username;
+    public String password;
 
-    public boolean login(String password, Database db) {
-        if (!db.user_accounts.containsKey(this.username)) return false;
+    //Constructor for Customer object.
+    public Customer(String name, int age) {
+        this.name = name;
+        this.age = age;
+    }
 
-        else if (db.user_accounts.get(this.username) != password) return false;
-        this.auth = true;
-        System.out.println("Authenticated succesfully");
-        bookings = new Ticket[100];
-        return true;
+    public Customer(String name, int age, String username, String password) {
+        this.name = name;
+        this.age = age;
+        this.username = username;
+        this.password = password;
     }
 
-    public boolean createAccount(Database db) throws Exception {
+
+    public boolean login(CustomerController c) throws IOException {
+        int tries = 9;
+        String tempUsername;
+        String tempPassword;
         BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+
+        System.out.println("Please enter your login username: ");
+        tempUsername = reader.readLine();
+        System.out.println("Please enter your password: ");
+        tempPassword = reader.readLine();
+
+        ArrayList<Customer> customers = null;
+        customers = c.getCustomerFromDB();  //Fetch all customers details from DB
+        for (int i = 0; i < customers.size(); i++) {
+            if (customers.get(i).getUsername() == tempUsername && customers.get(i).getPassword() == tempPassword) {
+                System.out.println("Authenticated successfully");
+                auth = true;
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean createAccount(CustomerController c) throws IOException{
         int tries = 9;  //9 tries before system shuts;
+        String tempUsername;
+        String tempPassword;
+        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
         do {
             System.out.println("Enter username: ");
-            String username = reader.readLine();
-            if (db.user_accounts.containsKey(username)) System.out.println("Username already taken");
+            tempUsername = reader.readLine();
+            if (c.checkUsernameExist(tempUsername)) System.out.println("Username already taken");
             if (tries == 0) System.out.println("Too many tries. System quitting now");
             tries--;
-        }
-        while (db.user_accounts.containsKey(username) && tries != 0);
+        } while (c.checkUsernameExist(tempUsername) && tries != 0);
         if (tries == 0) return false;
+        setUsername(tempUsername);
```

Comment:
I can't really find the setUserName method defined anywhere, in the Customer class or the CustomerController class. Are yu sure you've implemented this?

Reply:
setUsername and setPassword is under the Customer class near the bottom.